Packs/QRadar/Integrations/QRadar_v2/QRadar_v2.py

```diff
@@ -0,0 +1,1961 @@
+                start_offense_id += client.offenses_per_fetch
+            else:
+                latest_offense_fnd = True
+    return raw_offenses, fetch_query
+
+
+def fetch_incidents_long_running_samples():
+    last_run = get_integration_context(SYNC_CONTEXT)
+    return last_run.get("samples", [])  # type: ignore [attr-defined]
+
+
+def is_reset_triggered(lock, handle_reset=False):
+    """
+    Returns if reset signal is set. If handle_reset=True, will also reset the integration context
+    """
+    if lock.acquire(timeout=LOCK_WAIT_TIME):
+        ctx = get_integration_context(SYNC_CONTEXT)
+        if RESET_KEY in ctx:
+            if handle_reset:
+                print_debug_msg("Reset fetch-incidents.")
+                set_integration_context(
+                    {"samples": ctx.get("samples", [])}, sync=SYNC_CONTEXT
+                )
+            lock.release()
+            return True
+        lock.release()
+    return False
+
+
+def fetch_incidents_long_running_events(
+    client: QRadarClient,
+    user_query,
+    full_enrich,
+    fetch_mode,
+    events_columns,
+    events_limit,
+):
+    last_run = get_integration_context(SYNC_CONTEXT)
+    offense_id = last_run["id"] if last_run and "id" in last_run else 0
+
+    raw_offenses = fetch_raw_offenses(client, offense_id, user_query)
+
+    if len(raw_offenses) == 0:
+        return
+    if isinstance(raw_offenses, list):
+        raw_offenses.reverse()
+    for offense in raw_offenses:
+        offense_id = max(offense_id, offense["id"])
+    enriched_offenses = []
+
+    with concurrent.futures.ThreadPoolExecutor() as executor:
+        futures = []
+        for offense in raw_offenses:
+            futures.append(
+                executor.submit(
+                    enrich_offense_with_events,
+                    client=client,
+                    offense=offense,
+                    fetch_mode=fetch_mode,
+                    events_columns=events_columns,
+                    events_limit=events_limit,
+                )
+            )
+        for future in concurrent.futures.as_completed(futures):
+            enriched_offenses.append(future.result())
+
+    if is_reset_triggered(client.lock, handle_reset=True):
+        return
+
+    enriched_offenses.sort(key=lambda offense: offense.get("id", 0))
+    if full_enrich:
+        print_debug_msg("Enriching offenses")
+        enrich_offense_result(client, enriched_offenses, full_enrich)
+        print_debug_msg("Enriched offenses successfully.")
+    new_incidents_samples = try_create_incidents(enriched_offenses)
+    incidents_batch_for_sample = (
+        new_incidents_samples if new_incidents_samples else last_run.get("samples", [])
+    )
+
+    context = {LAST_FETCH_KEY: offense_id, "samples": incidents_batch_for_sample}
+    set_integration_context(context, sync=SYNC_CONTEXT)
+
+
+def try_create_incidents(enriched_offenses):
```

Comment:
why this is "try"?

Reply:
It's an older name that previously made sense - I'll change this to suit the current logic.